src/app/content/selectors.ts

```diff
@@ -20,6 +21,12 @@ export const book = createSelector(
   (state) => state.book
 );
 
+export const buyBookLink = createSelector(
+  book,
+  // ts is throwing an error because amazon_link is undefined in ArchiveBook
+  (data) => data && (data as BookWithOSWebData).amazon_link
```

Comment:
yea, there is a guard. you could change this to 
```suggestion
  (data) => hasOSWebData(data) ? data.amazon_link : undefined
```
or something like that